pkg/scheduler/framework/types.go

```diff
@@ -90,14 +90,16 @@ type ClusterEventWithHint struct {
 
 // QueueingHintFn returns a hint that signals whether the event can make a Pod,
 // which was rejected by this plugin in the past scheduling cycle, schedulable or not.
-// It's called before a Pod gets moved from unschedulableQ to backoffQ or activeQ.
+// It's called before a Pod gets moved from unschedulableQ to backoffQ or activeQ. And
+// If it returns an error, the caller should treat the event as QueueAfterBackoff so that
+// we can prevent the bug from resulting in the Pod stuck in the unschedulable pod pool.
```

Comment:
```suggestion
// It's called before a Pod gets moved from unschedulableQ to backoffQ or activeQ.
// If it returns an error, the scheduling queue behaves like QueueingHintFn returned QueueAfterBackoff so that
// we can prevent the Pod from stucking in the unschedulable pod pool.
```